src/ifaddrs.rs

```diff
@@ -42,11 +42,43 @@ cfg_if! {
     }
 }
 
+/// Workaround a bug in XNU where netmasks will always have the wrong size in
+/// the sa_len field due to the kernel ignoring trailing zeroes in the structure
+/// when setting the field. See https://github.com/nix-rust/nix/issues/1709#issuecomment-1199304470
+///
+/// To fix this, we overwrite `sa_len` to have the known (static) value of the
+/// structure.
+#[cfg(target_os = "macos")]
```

Comment:
Does the bug affect iOS too?

Reply:
I suppose so, given the kernel is the same, but I have no way to test - I don't own an iDevice.